scripts/zephyr_module.py

```diff
@@ -71,44 +89,79 @@ def process_module(module, cmake_out=None, kconfig_out=None):
-                     .format(module.yml, kconfig_setting))
+
+def process_cmake(module, meta):
+    section = meta.get('build', dict())
+    module_yml = os.path.join(module, 'zephyr/module.yml')
+    cmake_setting = section.get('cmake', None)
+    if not validate_setting(cmake_setting, module, 'CMakeLists.txt'):
+        sys.exit('ERROR: "cmake" key in {} has folder value "{}" which '
+                    'does not contain a CMakeLists.txt file.'
+                    .format(module_yml, cmake_setting))
 
     cmake_path = os.path.join(module, cmake_setting or 'zephyr')
     cmake_file = os.path.join(cmake_path, 'CMakeLists.txt')
-    if os.path.isfile(cmake_file) and cmake_out is not None:
-        cmake_out.write('\"{}\":\"{}\"\n'
+    if os.path.isfile(cmake_file):
+        return('\"{}\":\"{}\"\n'
                         .format(os.path.basename(module), PurePath(
                             os.path.abspath(cmake_path)).as_posix()))
+    else:
+        return ""
+
+def process_kconfig(module, meta):
+    section = meta.get('build', dict())
+    module_yml = os.path.join(module, 'zephyr/module.yml')
+    kconfig_setting = section.get('kconfig', None)
+    if not validate_setting(kconfig_setting, module):
+        sys.exit('ERROR: "kconfig" key in {} has value "{}" which does '
+                    'not point to a valid Kconfig file.'
+                    .format(module_yml, kconfig_setting))
+
 
     kconfig_file = os.path.join(module, kconfig_setting or 'zephyr/Kconfig')
-    if os.path.isfile(kconfig_file) and kconfig_out is not None:
-        kconfig_out.write('osource "{}"\n\n'
+    if os.path.isfile(kconfig_file):
+        return('osource "{}"\n\n'
                           .format(PurePath(
                               os.path.abspath(kconfig_file)).as_posix()))
+    else:
+        return ""
 
+def process_sanitycheck(module, meta):
 
-def main():
-    kconfig_out_file = None
-    cmake_out_file = None
+    out = ""
+    tests = meta.get('tests', dict())
+    samples = meta.get('samples', dict())
+    boards = meta.get('boards', dict())
+
+    for section in [tests, samples]:
+        pth =  section.get("path", None)
+        if pth:
+            dir = os.path.join(module, pth)
+            out += '-T\n{}\n'.format(PurePath(os.path.abspath(dir)).as_posix())
+
+    pth =  boards.get("path", None)
+    if pth:
+        dir = os.path.join(module, pth)
+        out += '--board-root\n{}\n'.format(PurePath(os.path.abspath(dir)).as_posix())
 
+    return out
+
+
+def main():
     parser = argparse.ArgumentParser(description='''
     Process a list of projects and create Kconfig / CMake include files for
     projects which are also a Zephyr module''')
 
     parser.add_argument('--kconfig-out',
                         help='File to write with resulting KConfig import'
                              'statements.')
+    parser.add_argument('--sanitycheck-out',
+                        help='File to write with resulting sanitycheck'
```

Comment:
I think you're just copy pasting the style from other arguments, but there is a missing space.  It should be `sanitycheck '` instead.